geoh5py/groups/property_group.py

```diff
@@ -75,32 +73,34 @@ class PropertyGroup:
     def __init__(  # pylint: disable=too-many-arguments
         self,
         parent: ObjectBase,
-        name: str | None = None,
+        association: str | DataAssociationEnum | None = None,
+        allow_delete: bool = True,
+        name: str = "Property Group",
         on_file: bool = False,
         uid: UUID | None = None,
-        property_group_type: GroupTypeEnum = GroupTypeEnum.UNKNOWN,
-        **kwargs,
+        property_group_type: GroupTypeEnum | str = GroupTypeEnum.UNKNOWN,
+        properties: list[UUID | Data] | None = None,
+        **_,
     ):
-        self._on_file: bool = False
-        self._allow_delete = True
-        self._association: DataAssociationEnum | None = None
-        self._properties: list[UUID] | None = None
+        self._parent: ObjectBase = self.validate_parent(parent)
+        self._property_group_type = self.validate_group_type(property_group_type)
```

Comment:
those functions could be private? (No telling it should just asking)

Reply:
Yep